Add login form validation

derick-montague opened the pull request:
- Sending incorrect credentials returns a 401 and we don't want the page
to redirect if we are trying to login. Wrapped the redirect in an if
block.
- Returning a promise used by the logout  action, which is needed
when not redirecting the page.  Didn't add to the if block since
other errors that use the router to redirect will need the Promise
returned also, e.g. 403.

derick-montague committed:
Add login form validation

- Sending incorrect credentials returns a 401 and we don't want the page
to redirect if we are trying to login. Wrapped the redirect in an if
block.
- Returning a promise used by the logout  action, which is needed
when not redirecting the page.  Didn't add to the if block since
other errors that use the router to redirect will need the Promise
returned also, e.g. 403.

derick-montague commented:
@yoshiemuranaka and @dixsie could you give this one a look when you have some time, please? :)

gtmills commented on src/views/Login/Login.vue:
```diff
@@ -50,18 +69,54 @@ export default {
+      this.errorMsg.title = null;
+      this.errorMsg.action = null;
+
+      // Name error message
```
update this comment to be more correct? Or remove?

derick-montague replied:
Removed in new commit.

gtmills commented on src/views/Login/Login.vue:
```diff
@@ -50,18 +69,54 @@ export default {
+            this.$router.push('/');
+          })
+          .catch(error => {
+            this.errorMsg.title = 'Inavalid username or password.';
```
* Invalid

derick-montague replied:
Resolved in new commit

gtmills commented:
Good stuff. looks reasonable to me.

derick-montague committed:
Remove unnecessary JavaScript comment

derick-montague committed:
Fix typo in error message text

yoshiemuranaka commented on src/views/Login/Login.vue:
```diff
@@ -50,18 +69,53 @@ export default {
+        password: null
+      },
+      disableSubmitButton: false,
+      submitButtonText: 'Login'
```
are you using this?

derick-montague replied:
Removed submitButtonText for now. The intent was to provide a message that the form was processing, but we can determine that later. Nice catch!

yoshiemuranaka commented on src/views/Login/Login.vue:
```diff
@@ -31,14 +45,19 @@
+            type="submit"
+            :disabled="disableSubmitButton"
+            variant="primary"
+            >Login</b-button
```
nitpick, can we change this to Log in? `Login`  = noun , `Log in` = verb

yoshiemuranaka commented on src/views/Login/Login.vue:
```diff
@@ -50,18 +69,53 @@ export default {
+        this.errorMsg.title = 'Username and password required.';
+      }
+
+      if (!this.errorMsg.title) {
```
Was there a reason for this extra if? versus returning false after line 90?

derick-montague committed:
Clean up code

- Update Login to Log in
- Restructure validate Form logic
- Use $spacer for margin in error message
- Remove unused data object property

derick-montague commented:
Migrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28757
